functions/utils/getToken.ts

```diff
@@ -11,12 +12,30 @@ function formatTitleName(symbol: string, name: string) {
   return 'View Token on Uniswap'
 }
 
-export default async function getToken(networkName: string, tokenAddress: string | undefined, url: string) {
+const convertTokenAddress = (tokenAddress: string, networkName: string) => {
+  if (tokenAddress === 'NATIVE') {
+    switch (networkName) {
+      case Chain.Celo:
+        return '0x471EcE3750Da237f93B8E339c536989b8978a438'
+      case Chain.Polygon:
+        return '0x0000000000000000000000000000000000001010'
+      default:
+        return undefined
+    }
+  }
+  return tokenAddress
+}
+
+export default async function getToken(networkName: string, tokenAddress: string, url: string) {
+  const origin = new URL(url).origin
+  const image = origin + '/api/image/tokens/' + networkName + '/' + tokenAddress
+  const uppercaseNetworkName = networkName.toUpperCase()
```

Comment:
Why is upper-casing necessary?

Reply:
Chain stores the names as completely upper case

Reply:
leave a comment explaining that